libs/platform/src/lib/components/form/form-group/form-group.component.ts

```diff
@@ -348,126 +418,50 @@ export class FormGroupComponent implements FormGroupContainer, OnInit, AfterCont
-     * and assign each field an rank number they will appear in teh UI. Just think of one columns
-     * layout when they are merged.
-     *
-     * [Field 1] - zoneLeft, rank 1
-     * [Field 2] - zoneLeft, rank 2
-     * [Field 3] - zoneRight, rank 20
-     * [Field 4] - zoneRight, rank 30
-     *
-     *
-     * [Field 1]      [Field 3]
-     * [Field 2]      [Field 4]
-     *
-     * When they are merged together into one column:
-     *
-     * [Field 1]
-     * [Field 3]
-     * [Field 2]
-     * [Field 4]
-     *
-     * We can pre-calculate and cache.
-     *
-     * We take a rank for each zone sort fields inside and then based on number of support it columns
-     * we need rearrange entries and give it different rank as it shows on the UI screen.
-     *
-     * Rank is important here, because its read by [style.order] in the template that nicelly
-     * re-arrange fields based on its rank number. Its pure CSS solution
-     *
-     *
-     * Todo: this should be more generic with variable num of columns
+     * @hidden
+     * if `columnLayoutType` is given, set those column layouts appropriately. Otherwise a layout will set on 2 columns
      */
-    private calculateMainZone(left: GroupField[], right: GroupField[]): GroupField[] {
-        if (left.length > 0 && right.length > 0) {
-            const merged: GroupField[] = [];
-            let indexL = 0,
-                indexR = 0,
-                current = 0;
-
-            while (current < left.length + right.length) {
-                if (indexL < left.length) {
-                    const f = new GroupField(
-                        left[indexL].zone,
-                        left[indexL].name,
-                        current,
-                        left[indexL].renderer,
-                        left[indexL].columns,
-                        left[indexL].isFluid
-                    );
-
-                    merged[current++] = f;
-                    indexL++;
-
-                    if (f.isFluid) {
-                        continue;
-                    }
-                    f.styleClass = `col-sm-${f.columns} col-md-${f.columns} col-lg-${f.columns}`;
-                }
-
-                if (indexR < right.length) {
-                    if (right[indexR].columns + left[indexL - 1].columns !== 12) {
-                        right[indexR].columns = 12 - left[indexL - 1].columns;
-                    }
-                    const f = new GroupField(
-                        right[indexR].zone,
-                        right[indexR].name,
-                        current,
-                        right[indexR].renderer,
-                        right[indexR].columns
-                    );
-
-                    f.styleClass = `col-sm-${f.columns} col-md-${f.columns} col-lg-${f.columns}`;
-                    merged[current++] = f;
-                    indexR++;
+    private _setUserLayout(): void {
```

Comment:
Where do you have the validation?   that your columnLayout is using smaller numbers than your column? We should also throw exception in dev mode

Reply:
Added separate method